From 445b93ae11c0c047b809ea4d79fd0bef88e3ea89 Mon Sep 17 00:00:00 2001
Message-Id: <445b93ae11c0c047b809ea4d79fd0bef88e3ea89.1398812352.git.mark.gross@intel.com>
In-Reply-To: <e87e3533536ae5eec0e06ceb7ef7428e42855790.1398812350.git.mark.gross@intel.com>
References: <e87e3533536ae5eec0e06ceb7ef7428e42855790.1398812350.git.mark.gross@intel.com>
From: Andy Ross <andrew.j.ross@intel.com>
Date: Tue, 18 Mar 2014 09:28:47 -0700
Subject: [PATCH 397/429] SQUISH: acpi-gpiobtn: fix initialization order

When built as =y, must be sure this gets initialized after all the
GPIO controllers.

Squish with:

    9a173c3 acpi-gpiobtn: GPIO Tablet buttons device

Change-Id: Ida0807c20b5da303fa1ddd258a63eae75a3a4091
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
Reviewed-on: https://android.intel.com/172709
Reviewed-by: Bergeron, Michael <michael.bergeron@intel.com>
Tested-by: Uyyala, Sridhar <sridhar.uyyala@intel.com>
Reviewed-by: Zhang, Jianxun <jianxun.zhang@intel.com>
Reviewed-by: Uyyala, Sridhar <sridhar.uyyala@intel.com>
---
 drivers/input/keyboard/acpi-gpiobtn.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/acpi-gpiobtn.c b/drivers/input/keyboard/acpi-gpiobtn.c
index b921bb7..8b2f088 100644
--- a/drivers/input/keyboard/acpi-gpiobtn.c
+++ b/drivers/input/keyboard/acpi-gpiobtn.c
@@ -128,5 +128,9 @@ static void __exit gpiobtn_exit(void)
 	platform_driver_unregister(&gpiobtn_plat_drv);
 }
 
-module_init(gpiobtn_init);
+/* The nature of the gpiolib API requires that all GPIO controllers be
+ * initialized before our probe routine so we can query the relevant
+ * gpio_desc objects. */
+late_initcall(gpiobtn_init);
+
 module_exit(gpiobtn_exit);
-- 
1.8.3.2

